client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/AbstractBinaryFormatReader.java

```diff
@@ -737,4 +732,19 @@ public ClickHouseBitmap getClickHouseBitmap(int index) {
     public void close() throws Exception {
         input.close();
     }
+
+    @Override
+    public int getColumnCount() {
+        return currentRecord.size();
+    }
+
+    @Override
+    public Set<String> getColumnNames() {
+        return currentRecord.keySet();
+    }
+
+    @Override
+    public Map<String, Object> current() {
+        return currentRecord;

```

Comment:
This cannot be exposed because it is internal records and should not be accessed in such way. I plan to change internal later.